lightkurve/lightcurve.py

```diff
@@ -1703,6 +1708,120 @@ def errorbar(self, **kwargs):
             ax.set_xlabel("Phase")
         return ax
 
+    def waterfall(self, ax=None, bin_points=1, phmin=-0.5, phmax=0.5, bin_method='mean',
+                        waterfall_method='average', **kwargs):
+        """Plot the folded light curve as a waterfall plot. Keywords are passed
+        to matplotlib.
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`
+            The matplotlib axes object.
+        bin_points : int
+            How many points should be in each bin
+        phmin : float
+            The minimum phase to plot
+        phmax : float
+            The maximum phase to plot
+        bin_method : str
+            The bin method. Choose from `'mean'` or `'median'`.
+        waterfall_method : 'str'
+            Type of waterfall plot, choose from `'average'` or `'sigma'`.
+            If `'average'`, the plot will display the average value in each bin.
+            If `'sigma'`, the plot will display the average in the bin divided by
+            the error in each bin, in order to show the data in terms of standard
+            deviation.
+        kwargs : dict
+            Dictionary of arguments to be passed to matplotlib
+        Returns
+        -------
+        ax : matplotlib.axes._subplots.AxesSubplot
+            The matplotlib axes object.
+        """
+        bin_method = validate_method(bin_method, supported_methods=['mean', 'median'])
+        if bin_method == 'mean':
+            bin_func = lambda y, e: (np.nanmean(y), np.sum(e**2)**0.5/len(e))
+        elif bin_method == 'median':
+            bin_func = lambda y, e: (np.nanmedian(y), np.sum(e**2)**0.5/len(e))
+
+        waterfall_method = validate_method(waterfall_method, supported_methods=['average', 'sigma'])
+        if waterfall_method == 'sigma':
+            waterfall_func = lambda y, e: (y - 1)/e
+        elif waterfall_method == 'average':
+            waterfall_func = lambda y, e: y
+        else:
+            raise ValueError('`waterfall_method` not understood.')
+
+        x, y, e = self.time_original[np.argsort(self.time_original)], self.flux[np.argsort(self.time_original)], self.flux_err[np.argsort(self.time_original)]
+        e /= np.nanmedian(self.flux)
+        y /= np.nanmedian(self.flux)
+
+        n = int(self.period/np.median(np.diff(x)) * (phmax - phmin)/bin_points)
```

Comment:
Should we use `nanmedian` here for extra safety?
```suggestion
        n = int(self.period/np.nanmedian(np.diff(x)) * (phmax - phmin)/bin_points)
```